### PR TITLE
Allow Array unknowns in OptimizationSystem

### PR DESCRIPTION
**Pull request by Vaibhavdixit02**
## Checklist

- [ ] Appropriate tests were added
- [ ] Any code changes were done in a way that does not break public API
- [ ] All documentation related to code changes were updated
- [ ] The new code follows the
  [contributor guidelines](https://github.com/SciML/.github/blob/master/CONTRIBUTING.md), in particular the [SciML Style Guide](https://github.com/SciML/SciMLStyle) and
  [COLPRAC](https://github.com/SciML/COLPRAC).
- [ ] Any new documentation only uses public API

### CODE REVIEW

File: src/systems/optimization/optimizationsystem.jl [hunk 1]
```diff
@@ -26,7 +26,7 @@ struct OptimizationSystem <: AbstractOptimizationSystem
     """Objective function of the system."""
     op::Any
     """Unknown variables."""
-    unknowns::Vector
```
Comment (ChrisRackauckas) [inline src/systems/optimization/optimizationsystem.jl:29]: I'm not sure how this becomes an array? I guess we just vec it. But how do we get to this?

Comment (Vaibhavdixit02) [reply src/systems/optimization/optimizationsystem.jl:29]: I am not sure I understand. This will allow us to handle matrix variables that's all

Comment (ChrisRackauckas) [reply src/systems/optimization/optimizationsystem.jl:29]: How did we get a non-vector array?

Comment (Vaibhavdixit02) [reply src/systems/optimization/optimizationsystem.jl:29]: You pass it in as the initial guess for the variables

**Pull request merged**